pkg/apis/maps/v1alpha1/webhook.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v1alpha1
+
+import (
+	"errors"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	ulog "github.com/elastic/cloud-on-k8s/pkg/utils/log"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+var (
+	groupKind     = schema.GroupKind{Group: GroupVersion.Group, Kind: Kind}
+	validationLog = ulog.Log.WithName("maps-v1alpha1-validation")
+
+	defaultChecks = []func(*ElasticMapsServer) field.ErrorList{
+		checkNoUnknownFields,
+		checkNameLength,
+		checkSupportedVersion,
+	}
+)
+
+// +kubebuilder:webhook:path=/validate-ems-k8s-elastic-co-v1alpha1-mapsservers,mutating=false,failurePolicy=ignore,groups=maps.k8s.elastic.co,resources=mapsservers,verbs=create;update,versions=v1alpha1,name=elastic-ems-validation-v1alpha1.k8s.elastic.co,sideEffects=None,admissionReviewVersions=v1alpha1,matchPolicy=Exact
+
+var _ webhook.Validator = &ElasticMapsServer{}
+
+func (m *ElasticMapsServer) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(m).
+		Complete()
+}
+
+func (m *ElasticMapsServer) ValidateCreate() error {
+	validationLog.V(1).Info("Validate create", "name", m.Name)
+	return m.validate(nil)
+}
+
+func (m *ElasticMapsServer) ValidateDelete() error {
+	validationLog.V(1).Info("Validate delete", "name", m.Name)
+	return nil
+}
+
+func (m *ElasticMapsServer) ValidateUpdate(old runtime.Object) error {
+	validationLog.V(1).Info("Validate update", "name", m.Name)
+	oldObj, ok := old.(*ElasticMapsServer)
+	if !ok {
+		return errors.New("cannot cast old object to ElasticMapsServer type")
+	}
+
+	return m.validate(oldObj)
+}
+
+func (m *ElasticMapsServer) validate(_ *ElasticMapsServer) error {
```

Comment:
nit: any reason for keeping this function's argument if there is no update check ? (or maybe update checks are missing ?)
```suggestion
func (m *ElasticMapsServer) validate() error {
```

Reply:
There is no upgrade check because there is currently no constraint on upgrades (you can even downgrade if you are so inclined). I will have a look if I can not just remove the whole bit.